backend/corpora/lambdas/api/v1/collection.py

```diff
@@ -110,23 +115,43 @@ def get_collection_dataset(dataset_uuid: str):
 
 @dbconnect
 def delete_collection(collection_uuid: str, visibility: str, user: str):
-    if visibility != CollectionVisibility.PRIVATE.name:
-        # Only allowed to delete private collections
-        return "", 405
     db_session = g.db_session
-    priv_collection = Collection.get_collection(
-        db_session,
-        collection_uuid,
-        CollectionVisibility.PRIVATE.name,
-        owner=_owner_or_allowed(user),
-        include_tombstones=True,
-    )
-    if priv_collection:
-        if not priv_collection.tombstone:
-            priv_collection.delete()
-        return "", 204
+    if visibility == CollectionVisibility.PUBLIC.name:
+        pub_collection = Collection.get_collection(
+            db_session,
+            collection_uuid,
+            visibility,
+            owner=_owner_or_allowed(user),
+            include_tombstones=True,
+        )
+        priv_collection = Collection.get_collection(
```

Comment:
Will this catch if there have been more than one attempted revision of the collection?
i.e.
1. Active revision
1. Tombstoned revision
1. Tombstoned revision

Is it possible hat `.get_collection()` could grab one tombstoned revision and skip:
```
if not priv_collection.tombstone:
                    priv_collection.delete()
```

Reply:
You'll see on line 152 that private collections are deleted not tombstoned. So in this case there is only ever one active revision, and never any tombstoned revisions. So to answer your question, is it possible that `.get_collection()` could grab one tombstone revision... I don't see how it could since no other revisions should exist in the database.